drivers/entropy/entropy_stm32.c

```diff
@@ -114,19 +119,49 @@ static int entropy_stm32_got_error(RNG_TypeDef *rng)
 	return 0;
 }
 
+#if defined(STM32_CONDRST_SUPPORT)
+static int recover_seed_error(RNG_TypeDef *rng)
+{
+	uint32_t count_timeout = 0;
+
+	LL_RNG_EnableCondReset(rng);
+	LL_RNG_DisableCondReset(rng);
+	/* When reset process is done cond reset bit is read 0
+	 * This typically takes: 2 AHB clock cycles + 2 RNG clock cycles.
+	 */
+
+	while (LL_RNG_IsEnabledCondReset(rng) ||
```

Comment:
Thanks very much for this feedback. I checked for SEIS in an earlier draft but removed it because the code never uses SEIS and the interrupt is masked. 

Reading the RM of stm32wl the following is not completely clear to me: Assuming a seed error occurs setting both SEIS and SECS. Would it be possible that the RNG recovers a seed error(SECS reset to 0) without an explicit trigger of CONDRST but there is still data with reduced entropy in the pipeline? Without knowing this for sure it would be probably safer to check for SEIS instead and initiate a soft reset. Can you confirm that if data is ready and SECS is 0, we can be sure the data we read has enough entropy?

Pretty much the same question applies to SOCS without CONDRST support.
The RM suggests to discard 12 words, in order to flush the buffer, as a recovery procedure after initialization.
In case it would be possible that there is data with reduced entropy in the pipeline while SECS is 0, I believe we should add the recovery described in the RM (e.g. RM0444 19.3.7) to recover_seed_error().

It would be great if you have more insight on this, just to be on the safe side. Or is this already ruled out in the RM and I have just overlooked it? I will add the check on SEIS as 2nd one, but I believe the sequence is already respected.

Reply:
> Would it be possible that the RNG recovers a seed error(SECS reset to 0) without an explicit trigger of CONDRST but there is still data with reduced entropy in the pipeline?
> Yes
Thanks for checking this.
I have now updated the code to check SEIS instead of SECS and either make a soft reset if available or read 12 words otherwise.